### ISSUE TITLE
Rule Request: vue/order-in-components LIFECYCLE_HOOKS ordering

### ISSUE THREAD

**Issue by webdevnerdstuff** [vuejs/eslint-plugin-vue#2075]
**What rule do you want to change?**
vue/order-in-components

**Does this change cause the rule to produce more or fewer warnings?**
More. It will help to create consistency when working on projects in teams where people will either forgot, or do not follow style guides like they should.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Adjusting the option for `LIFECYCLE_HOOKS`. For example instead of just `LIFECYCLE_HOOKS`, have it order them in the defined order on Line 80 of this file:
https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/order-in-components.js

Or in addition to, allow replacing `LIFECYCLE_HOOKS` with a supplied order. Ex.
```
[
    'beforeCreate',
    'created',
    'beforeMount',
    'mounted',
    'beforeUpdate',
    'updated',
    'activated',
    'deactivated',
    'beforeUnmount', // for Vue.js 3.x
    'unmounted', // for Vue.js 3.x
    'beforeDestroy',
    'destroyed',
    'renderTracked', // for Vue.js 3.x
    'renderTriggered', // for Vue.js 3.x
    'errorCaptured' // for Vue.js 2.5.0+
  ]
```

**Please provide some example code that this change will affect:**
For example, this is not in the recommended order of [Vue Lifecycle Events](https://vuejs.org/guide/essentials/lifecycle.html#lifecycle-diagram), in the order they are called. But it does not get picked up by the linter as it currently is and considers it valid (no errors/warnings/etc).

```vue
<script>
export default {
	name: 'FooBar',
	renderTriggered() {
		// do something
	},
	renderTracked() {
		// do something
	},
	errorCaptured() {
		// do something
	},
	unmounted() {
		// do something
	},
	beforeUnmount() {
		// do something
	},
	deactivated() {
		// do something
	},
	activated() {
		// do something
	},
	updated() {
		// do something
	},
	beforeUpdate() {
		// do something
	},
	beforeCreate() {
		// do something
	},
	mounted() {
		// do something
	},
	beforeMount() {
		// do something
	},
	created() {
		// do something
	},
};
</script>
```

**What does the rule currently do for this code?**
It currently only accounts for the order of the `LIFECYCLE_HOOKS` in the order list, but does not account for the order of the hooks inside of `LIFECYCLE_HOOKS` .

**What will the rule do after it's changed?**
Allow for order OF those lifecycle hooks.

**Additional context**
Example of the .eslintrc.js file I tested with:
```
module.exports = {
	env: {
		browser: true,
		es2021: true,
	},
	extends: [
		'plugin:vue/vue3-recommended',
		'eslint:recommended',
	],
	parser: 'vue-eslint-parser',
	plugins: [
		'import',
		'vue',
	],
	rules: {
		'vue/order-in-components': ['error', {
			'order': [
				'el',
				'name',
				'key',
				'parent',
				'functional',
				['delimiters', 'comments'],
				['components', 'directives', 'filters'],
				'extends',
				'mixins',
				['provide', 'inject'],
				'ROUTER_GUARDS',
				'layout',
				'middleware',
				'validate',
				'scrollToTop',
				'transition',
				'loading',
				'inheritAttrs',
				'model',
				['props', 'propsData'],
				'emits',
				'setup',
				'asyncData',
				'data',
				'fetch',
				'head',
				'computed',
				'watch',
				'watchQuery',
				'LIFECYCLE_HOOKS',
				'methods',
				['template', 'render'],
				'renderError'
			]
		}],
	},
};
```

**Comment by ota-meshi** [comment]
I think you can define the names of the lifecycle hooks in your preferred order in the rule option instead of the `LIFECYCLE_HOOKS` keyword.

**Comment by webdevnerdstuff** [comment]
I could have swore I tested that a few times before submitting this issue. *facepalm. Tested it out and is working like it should. Thank you for your response to catch my brain fart.

Perhaps the documentation would benefit by including these values to replace `LIFECYCLE_HOOKS`?
https://eslint.vuejs.org/rules/order-in-components.html#options

**Comment by ota-meshi** [comment]
I'm not sure how to describe it in the documentation so that it is clear to users, but please suggest it in the pull request 😉.
